docs/schema/extra.json

```diff
@@ -13,41 +13,78 @@
       "title": "Analytics provider",
       "markdownDescription": "https://squidfunk.github.io/mkdocs-material/setup/setting-up-site-analytics/#google-analytics",
       "type": "object",
-      "properties": {
-        "provider": {
-          "title": "Analytics provider",
-          "anyOf": [
-            {
-              "title": "Google Analytics",
-              "markdownDescription": "https://squidfunk.github.io/mkdocs-material/setup/setting-up-site-analytics/#google-analytics",
-              "enum": [
-                "google"
-              ]
-            },
-            {
-              "type": "string"
+      "allOf": [
+        {
+          "if": {
+            "properties": {
+              "provider": {
+                "enum": [
+                  "google"
+                ]
+              }
             }
-          ]
-        },
-        "property": {
-          "anyOf": [
-            {
-              "title": "Google Analytics 4",
-              "markdownDescription": "https://squidfunk.github.io/mkdocs-material/setup/setting-up-site-analytics/#google-analytics",
-              "pattern": "^G-\\w{10}$"
-            },
-            {
-              "title": "Universal Analytics",
-              "markdownDescription": "https://squidfunk.github.io/mkdocs-material/setup/setting-up-site-analytics/#google-analytics",
-              "pattern": "^UA-\\w{9}-\\w$"
+          },
+          "then": {
+            "properties": {
+              "provider": {
+                "title": "Google Analytics",
+                "markdownDescription": "https://squidfunk.github.io/mkdocs-material/setup/setting-up-site-analytics/#google-analytics",
+                "enum": [
+                  "google"
+                ]
+              },
+              "property": {
+                "anyOf": [
+                  {
+                    "title": "Google Analytics 4",
+                    "markdownDescription": "https://squidfunk.github.io/mkdocs-material/setup/setting-up-site-analytics/#google-analytics",
+                    "pattern": "^G-\\w{10}$"
+                  },
+                  {
+                    "title": "Universal Analytics",
```

Comment:
We can remove UA, as UA has been sunset and we don't support it anymore .

Reply:
I've pushed a separate commit for this because it's an independent change.